src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h

```diff
@@ -197,20 +234,23 @@ inline void postcomputeImposedStateFromNewStress(
 
   using namespace dr::misc::quantity_indices;
 
-#ifndef ACL_DEVICE_OFFLOAD
+#ifndef ACL_DEVICE
   checkAlignmentPostCompute(
       qIPlus, qIMinus, imposedStateP, imposedStateM, faultStresses, tractionResults);
 #endif
 
   for (unsigned o = 0; o < CONVERGENCE_ORDER; ++o) {
     auto weight = timeWeights[o];
 
-#ifdef ACL_DEVICE_OFFLOAD
-#pragma omp loop bind(parallel)
-#else
-#pragma omp simd
+    using NumPointsRange = typename NumPoints<Type>::Range;
+#ifndef ACL_DEVICE
+    #pragma omp simd
 #endif
-    for (unsigned i = 0; i < misc::numPaddedPoints; ++i) {
+    for (auto index = NumPointsRange::start; index < NumPointsRange::end;
+         index += NumPointsRange::step) {
+
+      auto i{startIndex + index};
```

Comment:
Also consider to put this into the loop header.

Reply:
discussed